libsplinter/src/rest_api/auth/authorization/maintenance/mod.rs

```diff
@@ -248,7 +248,7 @@ mod tests {
             unimplemented!()
         }
 
-        fn update_role(&self, _role: Role) -> Result<(), RoleBasedAuthorizationStoreError> {
+        fn update_role(&self, _role: Role) -> Result<bool, RoleBasedAuthorizationStoreError> {
```

Comment:
I wonder if it would make more sense to add a `ConstraintViolationType::NotFound` variant? If we decide to go with what you have here, we should probably update the data store guidelines to describe this approach.

Reply:
I'm fine with either approach.

Reply:
I've updated this to use the `ConstraintViolationType::NotFounnd`, which is also added in this PR.